### ISSUE TITLE
Feature Request: Add 'Accepted' Field to dev_data in JetBrains IDEA Plugin for Consistency with VSCode Plugin

### ISSUE THREAD

**Issue by loufigoGo** [continuedev/continue#2335]
### Validations

- [ ] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

In the JetBrains IDEA plugin, the locally generated ~/.continue/dev_data/devdata.sqlite only contains the tokens_generated table and lacks the accepted field, unlike the VSCode plugin. It is crucial for the IDEA plugin to also include the accepted field in the local dev_data.

**Comment by Patrick-Erichsen** [comment]
Thanks for flagging this @loufigoGo , I wasn't aware that we aren't logging the accepted field in the VS Code plugin. 

We handle dev data logging at the core level here: https://github.com/continuedev/continue/blob/bc9ab739519f696c6ab77ff5c3b5e7913bd0232c/core/autocomplete/completionProvider.ts#L206-L207

And for the JetBrains plugin, I believe this is where we are logging autocomplete acceptance: https://github.com/continuedev/continue/blob/bc9ab739519f696c6ab77ff5c3b5e7913bd0232c/extensions/intellij/src/main/kotlin/com/github/continuedev/continueintellijextension/autocomplete/AcceptAutocompleteAction.kt#L14-L15

For VS Code, it's here: https://github.com/continuedev/continue/blob/bc9ab739519f696c6ab77ff5c3b5e7913bd0232c/extensions/vscode/src/commands.ts#L565-L566

It looks like we aren't passing the `completionId` argument on JetBrains.

**Comment by loufigoGo** [comment]
Thank you for the response @Patrick-Erichsen . I'm really looking forward to JetBrains' dev_data including completion and accepted fields, and I hope vscode's dev_data can add timestamps too.  Could you please let me know when the version update might be available ?

**Comment by Patrick-Erichsen** [comment]
We've got a decent backlog of JetBrains related enhancements/fixes so I can't give a good estimate but I've made sure to include this in the list. 

Curious what your usecase for the timestamps are?

**Comment by Patrick-Erichsen** [comment]
@loufigoGo we fixed this here: https://github.com/continuedev/continue/pull/2394

Will be available in next pre-release!

Also, I should have mentioned this previously, but autocomplete logs are not written to `~/.continue/dev_data/devdata.sqlite`. They are written to `~/.continue/dev_data/autocomplete.jsonl`.

**Comment by loufigoGo** [comment]
Great, but I noticed that my .continue/dev_data still hasn't updated the autocomplete.jsonl with jetBrains-IDE. I have used the dev branch for packaging and compiling .

**Comment by loufigoGo** [comment]
In autocomplete.jsonl, I need the timestamps field to understand when users use autocomplete for easier filtering. Otherwise, I would have to check devdata.sqlite at the same time, which is quite inconvenient.

**Comment by loufigoGo** [comment]
`    fun accept() {
        val completion = pendingCompletion ?: return
        val text = completion.text ?: return
        val editor = completion.editor
        val offset = completion.offset
        editor.document.insertString(offset, text)
        editor.caretModel.moveToOffset(offset + text.length)
        **val input = mapOf(
            "completionId" to completion.completionId
        )**
        project.service<ContinuePluginService>().coreMessenger?.request("autocomplete/accept", input, null, ({}))
        invokeLater {
            clearCompletions(editor)
        }
    } ` it looks like need add a completion.completionId input with mapOf() to autocomplete/accept request ?

**Comment by glm1024** [comment]
Hello, We need timestamps as well. Our team uses this feature to evaluate how many autocomplete code suggestions generated and how many accepted each month.

**Comment by glm1024** [comment]
I have update continue to 0.0.75. But I noticed that my .continue/dev_data still hasn't updated the autocomplete.jsonl with jetBrains-IDE too.

**Comment by Patrick-Erichsen** [comment]
Thanks for the confirmation here @glm1024 - timestamps are a relatively easy add on to the logs as well. I'll plan to take a second look at this later this week.

**Comment by glm1024** [comment]
Copy that. Thanks so much for the response. Also, I wonder if I can become a contributor to this project ?

**Comment by Patrick-Erichsen** [comment]
@glm1024 definitely! If you're interested in adding the timestamps to the autocomplete logs that would actually be great. 

Here are a few resources to get you started:
- Contributing docs: https://github.com/continuedev/continue/blob/dev/CONTRIBUTING.md
- JetBrains specific contributing docs: https://github.com/continuedev/continue/blob/dev/extensions/intellij/CONTRIBUTING.md
- My previous comment mentioning the specific places we do autocomplete logging: https://github.com/continuedev/continue/issues/2335#issuecomment-2364528513

Feel free to follow up here if you have any questions/issues getting started.

**Comment by glm1024** [comment]
Okay，I will have a try. : ) @Patrick-Erichsen

**Comment by glm1024** [comment]
test

**Comment by glm1024** [comment]
test

**Comment by glm1024** [comment]
> Great, but I noticed that my .continue/dev_data still hasn't updated the autocomplete.jsonl with jetBrains-IDE. I have used the dev branch for packaging and compiling .

This issue has been fixed, notice next release please.

**Comment by glm1024** [comment]
> In autocomplete.jsonl, I need the timestamps field to understand when users use autocomplete for easier filtering. Otherwise, I would have to check devdata.sqlite at the same time, which is quite inconvenient.

Timestamp has been added, notice next release too. @loufigoGo

**Comment by glm1024** [comment]
> ### Validations
> * [ ]  I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
> * [x]  I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement
> 
> ### Problem
> In the JetBrains IDEA plugin, the locally generated ~/.continue/dev_data/devdata.sqlite only contains the tokens_generated table and lacks the accepted field, unlike the VSCode plugin. It is crucial for the IDEA plugin to also include the accepted field in the local dev_data.
> 
> ### Solution
> _No response_

This may be a little difficult, cause I found that the data is recorded when LLM generates the response, not when user press the tab key. How about useing autocomplete.jsonl instead ? @loufigoGo

**Comment by glm1024** [comment]
> Okay，I will have a try : ) @Patrick-Erichsen

I have fixed the issue, Mr. Patrick, and Mr. Nate has merged the code. @Patrick-Erichsen

**Comment by loufigoGo** [comment]
> > ### Validations
> > 
> > * [ ]   I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
> > * [x]   I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement
> > 
> > ### Problem
> > In the JetBrains IDEA plugin, the locally generated ~/.continue/dev_data/devdata.sqlite only contains the tokens_generated table and lacks the accepted field, unlike the VSCode plugin. It is crucial for the IDEA plugin to also include the accepted field in the local dev_data.
> > ### Solution
> > _No response_
> 
> This may be a little difficult, cause I found that the data is recorded when LLM generates the response, not when user press the tab key. How about useing autocomplete.jsonl instead ? @loufigoGo

@glm1024  great ! very thanks to you !

**Comment by glm1024** [comment]
The latest version still has a problem. Every accepted attribute in the file autocomplete.jsonl is set to true. I have given feedback to the Continue. We still need to wait.
